addons/mail/wizard/mail_compose_message.py

```diff
@@ -243,7 +243,7 @@ def send_mail(self, auto_commit=False):
                         batch_mails_sudo |= self.env['mail.mail'].sudo().create(mail_values)
                     else:
                         post_params = dict(
-                            message_type=wizard.message_type,
+                            message_type=str(wizard.message_type),
```

Comment:
You really want "False" ? Why is it required to stringify message_type from wizard ?

Reply:
The only valid values for the (required) Selection field of the wizard are `comment` and `notification`. This is sent to `message_post` that does this check : https://github.com/odoo/odoo/blob/834ebc26ff4bc30f370cdec2656d71612a6e8fab/addons/mail/models/mail_thread.py#L1832
Since `user_notification` is not a valid value, it would log a warning through the `__eq__` function of `SelectionStr`.

Reply:
Still not understanding why we have to stringify an already-stringified value, as you pointed out it was required.

Reply:
This is because of the changes in `fields.py` done in this PR. The string stored in the cache is actually an instance of a super-type of `str`, that checks that the values to which we compare it are valid selection values.

Reply:
Ouch, so much obscurity.  This sucks hard, really hard.

Reply:
Well maybe you should consider finding an implementation that does not require adding code to a standard string comparison... really not liking this.